src/box/alter.cc

```diff
@@ -1021,7 +1021,9 @@ alter_space_do(struct txn_stmt *stmt, struct alter_space *alter)
 
 	alter->new_space->sequence = alter->old_space->sequence;
 	alter->new_space->sequence_fieldno = alter->old_space->sequence_fieldno;
-	alter->new_space->sequence_path = alter->old_space->sequence_path;
+	if (alter->old_space->sequence_path != NULL)
+		alter->new_space->sequence_path =
+			(char *)xstrdup(alter->old_space->sequence_path);
```

Comment:
Let's split this patch into different commits, for example:
- Leak on DDL access check failure.
- Leak on xlog open failure.
- Leak on iproto override.
- Leak on replication reconfiguration.
- Leak on network address resolution error.

Also, it's preferable to submit some fixes in separate PRs because a PR with > 5-7 commits is difficult to review.